Cannot find name 'Optinal'

**Describe the bug**
A Typescript Error.

**To Reproduce**
Just installed the package.

**Error**
```ts
node_modules/@metaplex/js/lib/actions/shared/approve.d.ts:9:48 - error TS2304: Cannot find name 'Optional'.

9 export declare function createApproveTxs(args: Optional<CreateApproveParams, 'authority'>): {
                                                 ~~~~~~~~
```

**Notes**
I quickly fixed it by adding the declaration of `Optional` in the `@metaplex/js/lib/actions/shared/approve.d.ts` file.

Hitting the same issue. Using npm. Installing `"@metaplex/js": "^4.11.1",`

Can't build with this issue — do we run tsc builds as smoke tests before shipping? I wonder if there is way to prevent these issues from getting shipped to prod.

```JS
➜  wonka git:(main) npm install

> @triton-labs/wonka@1.0.3 prepare
> npm run build


> @triton-labs/wonka@1.0.3 build
> tsc

node_modules/@metaplex/js/lib/actions/shared/approve.d.ts:9:48 - error TS2304: Cannot find name 'Optional'.

9 export declare function createApproveTxs(args: Optional<CreateApproveParams, 'authority'>): {
                                                 ~~~~~~~~


Found 1 error.

npm ERR! code 2
npm ERR! path /Users/zkhalapyan/Desktop/wonka
npm ERR! command failed
npm ERR! command sh -c npm run build

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/zkhalapyan/.npm/_logs/2022-01-12T21_55_42_842Z-debug-0.log
```

Here is my package.json:

```js
➜  wonka git:(main) ✗ cat package.json
{
  "name": "@triton-labs/wonka",
  "version": "1.0.3",
  "description": "Wonka is a JS library meant that simplifies the minting process on Metaplex's CandyMachine. ",
  "main": "lib/index.js",
  "types": "lib/index.d.ts",
  "scripts": {
    "test": "jest --config jestconfig.json",
    "build": "tsc",
    "format": "prettier --write \"src/**/*.ts\" \"src/**/*.js\"",
    "lint": "tslint -p tsconfig.json",
    "prepare": "npm run build",
    "prepublishOnly": "npm test && npm run lint",
    "preversion": "npm run lint",
    "version": "npm run format && git add -A src",
    "postversion": "git push && git push --tags"
  },
  "repository": {
    "url": "https://github.com/TritonLabs/Wonka"
  },
  "keywords": [
    "candy",
    "machine",
    "metaplex",
    "nft",
    "solana",
    "crypto",
    "mint"
  ],
  "files": [
    "lib/**/*"
  ],
  "author": "@0xZoZoZo",
  "license": "ISC",
  "dependencies": {
    "@metaplex/js": "^4.11.1",
    "@project-serum/anchor": "^0.18.2",
    "@solana/spl-token": "^0.1.8",
    "@solana/wallet-adapter-base": "^0.9.2",
    "@solana/web3.js": "^1.30.2",
    "@testing-library/jest-dom": "^5.11.4",
    "@testing-library/react": "^11.1.0",
    "@testing-library/user-event": "^12.1.10",
    "arweave": "^1.10.18",
    "assert": "^2.0.0",
    "loglevel": "^1.8.0",
    "next": "^12.0.7",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-scripts": "^3.0.1",
    "web-vitals": "^1.0.1"
  },
  "devDependencies": {
    "@types/jest": "^27.4.0",
    "@webpack-cli/generators": "^1.1.0",
    "husky": "^7.0.4",
    "jest": "^27.4.7",
    "prettier": "^2.5.1",
    "ts-jest": "^27.1.2",
    "tslint": "^6.1.3",
    "tslint-config-prettier": "^1.18.0",
    "typescript": "^4.5.4",
    "webpack-cli": "^4.9.1"
  }
}
```

--- a/src/actions/shared/approve.ts
+++ b/src/actions/shared/approve.ts
@@ -1,5 +1,6 @@
 import { Token, TOKEN_PROGRAM_ID, u64 } from '@solana/spl-token';
 import { Keypair, PublicKey, Transaction } from '@solana/web3.js';
+import { Optional } from '../../types';
 
 interface CreateApproveParams {
   authority: Keypair;

--- a/src/types.ts
+++ b/src/types.ts
@@ -42,3 +42,5 @@ export type MetadataJson = {
   collection?: MetadataJsonCollection;
   properties: MetadataJsonProperties;
 };
+
+export type Optional<T, K extends keyof T> = Pick<Partial<T>, K> & Omit<T, K>;

--- a/src/typings/index.d.ts
+++ b/src/typings/index.d.ts
@@ -1,1 +1,0 @@
-type Optional<T, K extends keyof T> = Pick<Partial<T>, K> & Omit<T, K>;

